### PR TITLE
Fix errors found by schema validation (fixes #2779)

### LINKED ISSUE #2779: Fix issues found by schema validator
I've been working on the schema validator (#2562) so that we can track down and fix hard-to-find bugs. This issue stands to get feedback on some cases where I'm not sure what the correct path is, as well as to serve as a checklist for the validation.

For completeness, I intend to validate every campaign on every difficulty, as well as MP, the editor, and tests. To be clear, all I actually need to do is load each one up, look at the log, fix any errors, and repeat until there are no errors - I'm not going to be playing through any of the campaigns or anything. I don't expect to find any errors on medium or hard that aren't also on easy, so I'll probably just load medium and hard (and nightmare where applicable) once just to be sure.

* [x] Core WML (that which is loaded by simply getting to the title screen)
* [x] Editor WML
* [x] Interactive tests (accessible by hotkey)
* [x] Unit tests (not accessible by hotkey)
* [x] Multiplayer WML (that which is loaded by simply getting to the create screen)
* [x] Hornshark Island (2P)
* [x] A New Land (4P)
* [x] Team Survival (6P)
* [x] Legend of Wesmere MP
* [x] Tutorial
* [x] An Orcish Incursion
* [x] Dead Water
* [x] Defador's Memoirs
* [x] Descent into Darkness
* [x] Eastern Invasion
* [x] Hammer of Thursagan
* [x] Heir to the Throne
* [x] Legend of Wesmere
* [x] Liberty
* [x] Northern Rebirth
* [x] Rise of Wesnoth
* [x] Sceptre of Fire
* [x] Secrets of the Ancients
* [x] Son of the Black Eye
* [x] South Guard
* [x] Two Brothers
* [x] Under the Burning Suns
* [ ] Under the Burning Suns (old units) - I know they're going to be removed soon, but I figure checking them anyway could potentially help find bugs in the schema itself

### PR DESCRIPTION
**Pull request by CelticMinstrel**
Not done yet, but I'd like feedback on the LoW changes in particular. I may have made some incorrect choices there.

### CODE REVIEW

File: data/campaigns/Northern_Rebirth/scenarios/06a_Old_Friend.cfg [hunk 1]
```diff
@@ -289,13 +289,17 @@
         [if]
             [have_unit]
                 side=1
-                x=$starting_point.x
-                y=$starting_point.y
-                range=10
-                [not]
+                [filter_location]
```
Comment (gfgtdf) [inline data/campaigns/Northern_Rebirth/scenarios/06a_Old_Friend.cfg]: this coudl probaly be impoved to use location_id=1

**Comment by CelticMinstrel** [comment]
While this isn't completely finished, the most important part (the campaigns and MP WML) is basically done (once I push the TRoW fixes). So this can be opened up for review now.

File: data/campaigns/Northern_Rebirth/scenarios/06a_Old_Friend.cfg (hunk 1 above)
Comment (ln-zookeeper) [reply data/campaigns/Northern_Rebirth/scenarios/06a_Old_Friend.cfg]: What would that do? Looks like that feature is still undocumented, anyway.

File: data/campaigns/The_Rise_Of_Wesnoth/scenarios/16_The_Kalian.cfg [hunk 2]
```diff
@@ -42,11 +42,10 @@
 
     [side]
         side=3
-        type=Elvish Champion
+        type=Elvish Champion Kalian
```
Comment (ln-zookeeper) [inline data/campaigns/The_Rise_Of_Wesnoth/scenarios/16_The_Kalian.cfg]: How did this end up here? There's no such unit type.

File: data/campaigns/Northern_Rebirth/scenarios/06a_Old_Friend.cfg (hunk 1 above)
Comment (CelticMinstrel) [reply data/campaigns/Northern_Rebirth/scenarios/06a_Old_Friend.cfg]: `location_id=number` means "the starting location of that side".

File: data/campaigns/The_Rise_Of_Wesnoth/scenarios/16_The_Kalian.cfg (hunk 2 above)
Comment (gfgtdf) [inline data/campaigns/The_Rise_Of_Wesnoth/scenarios/16_The_Kalian.cfg]: why this change?

Comment (gfgtdf) [reply data/campaigns/The_Rise_Of_Wesnoth/scenarios/16_The_Kalian.cfg]: hmm ok. But having such a custom unit will make it incompatible with older clients, (since the unit_type data is not sended over network, if the newer client is host, older cleints will get a 'unknown unit type' error).  So an [object] based solution would really be preferrable if possible.

Comment (CelticMinstrel) [reply data/campaigns/The_Rise_Of_Wesnoth/scenarios/16_The_Kalian.cfg]: I guess `[effect]apply_to=image_mod` could work here by somehow blanking out the standard image and blitting on the new image, but that's extremely ugly... I think it would be the only way to get the effect with an `[object]`.

**Comment by CelticMinstrel** [comment]
Unless anyone has additional comments, this is ready to merge. I changed the Kalian council member in TRoW to use an `[object]`, as requested.

File: data/campaigns/Legend_of_Wesmere/scenarios/chapter5/23_End_of_War.cfg [hunk 3]
```diff
@@ -60,7 +60,6 @@
     [side]
         side=2
 
-        no_leader=yes
```
[the remark this answers is not in the document]
Comment (CelticMinstrel) [reply data/campaigns/Legend_of_Wesmere/scenarios/chapter5/23_End_of_War.cfg]: I can't quite remember, but most likely because of the `canrecruit=yes` and/or the `{PLAYABLE}` macro? Basically the idea is not to include keys for defining a leader unit when `no_leader=yes`, and not include `no_leader=yes` if there are keys defining a leader unit. Should I have removed the `canrecruit=yes` instead here?

**Comment by CelticMinstrel** [comment]
I've now partially merged this - all the fixes that are straight fixes have been merged to 1.14, while cleanup remains in this branch for review and possible later merge.

**Comment by CelticMinstrel** [comment]
Thanks for looking at it, @sigurdfdragon. Since you came in so late, you may have missed the portion that was already merged. If you'd like to take a look, you can browse backwards from [here](https://github.com/wesnoth/wesnoth/commit/ad7b0ef14059226fbbe711a271b35ec4925de89d) using GitHub's "parent commit" link.

**Comment by sigurdfdragon** [comment]
The merged ones seem good

**Comment by Vultraz** [comment]
If you resolve the conflicts this can be merged.

**Comment by CelticMinstrel** [comment]
Can't you resolve them yourself?

**Comment by CelticMinstrel** [comment]
Merged manually.
